### PR TITLE
git: add tag and push methods

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: libgiza/git.py [hunk 1]
```diff
@@ -202,6 +202,25 @@ def update(self):
...
+    def push(self, remote='origin', ref=None):
+        if ref is None:
+            ref = self.current_branch()
+
+        return self.cmd("push", remote, ref)
+
+    def tag(self, name, delete=False, force=False):
+        args = ["tag"]
+
+        if delete is True:
```
Comment (i80and) [inline libgiza/git.py]: PEP-8 strongly advises against this style, in favor of merely ``if delete:``

File: libgiza/git.py [hunk 2]
```diff
@@ -202,6 +202,25 @@ def update(self):
...
+            args.append("-t")
```
Comment (i80and) [inline libgiza/git.py]: What's "-t"? ``git tag`` doesn't seem to recognize it, unless I'm being daft.

Comment (tychoish) [inline libgiza/git.py]: should be `-d`

File: libgiza/git.py (hunk 1 above)
Comment (tychoish) [inline libgiza/git.py]: bleh, I remembered this admonition but in reverse, but it's all over giza/libgiza. maybe I'll change it in a larger sweep later?

Comment (i80and) [inline libgiza/git.py]: That's fine. The cases in which it matters are pretty esoteric.

**Pull request closed without merging**